### PR TITLE
Update to latest codecs API

### PR DESCRIPTION
**Pull request by davezuch**
## What does this pull request do?

This PR is dependent on [purescript-argonaut-codecs#47](https://github.com/purescript-contrib/purescript-argonaut-codecs/pull/47). That one introduces breaking changes that propagate here.

### CODE REVIEW

**Comment by thomashoneyman** [comment]
Hey @davezuch! Is this ready to be updated now that core has been?

**Comment by davezuch** [comment]
@thomashoneyman I [created a PR](https://github.com/purescript-contrib/purescript-argonaut-traversals/pull/23) for `purescript-argonaut-traversals` that'll need to be released before I can fix the dependencies for this repo

**Comment by thomashoneyman** [comment]
@davezuch I've released that version.

File: src/Data/Argonaut.purs [hunk 1]
```diff
@@ -9,7 +9,7 @@ module Data.Argonaut
   ) where
 
 import Data.Argonaut.Core (Json, caseJson, caseJsonArray, caseJsonBoolean, caseJsonNull, caseJsonNumber, caseJsonObject, caseJsonString, fromArray, fromBoolean, fromNumber, fromObject, fromString, isArray, isBoolean, isNull, isNumber, isObject, isString, jsonEmptyArray, jsonEmptyObject, jsonEmptyString, jsonFalse, jsonNull, jsonSingletonArray, jsonSingletonObject, jsonTrue, jsonZero, stringify, toArray, toBoolean, toNull, toNumber, toObject, toString)
-import Data.Argonaut.Decode (class DecodeJson, decodeJson, defaultField, getField, getFieldOptional, (.?), (.?=), (.??))
+import Data.Argonaut.Decode (class DecodeJson, decodeJson, getField, getFieldDeprecated, getFieldOptional, getFieldOptionalDeprecated, getFieldOptional', defaultField, defaultFieldDeprecated, (.:), (.?), (.:!), (.:?), (.??), (.!=), (.?=))
```
Comment (thomashoneyman) [inline src/Data/Argonaut.purs]: @davezuch looks like this isn't importing all combinators

**Pull request merged**